### PR TITLE
[16.0] automation_oca add dependencies to use today() and datetime in domain (filter)

### PR DESCRIPTION
**Pull request by gaelTorrecillas**
hello,

this pr purpose to fix on the use datetime in filter (domain) https://github.com/OCA/automation/issues/5

### CODE REVIEW

**Comment by flotho** [comment]
Hi @gaelTorrecillas thanks for this PR, any chance to fix the lint ?

**Comment by pedrobaeza** [comment]
Please follow commit message guidelines:

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message

File: automation_oca/models/automation_configuration.py [hunk 1]
```diff
@@ -5,7 +5,18 @@
...
+)
+from odoo.tools.safe_eval import (
+    dateutil as safe_dateutil,
+)
+from odoo.tools.safe_eval import (
+    safe_eval,
+)
+from odoo.tools.safe_eval import (
+    time as safe_time,
+)
```
Comment (pedrobaeza) [inline automation_oca/models/automation_configuration.py]:
```suggestion
from odoo.tools.safe_eval import (
    datetime as safe_datetime,
    dateutil as safe_dateutil,
    safe_eval,
    time as safe_time,
)

File: automation_oca/models/automation_configuration.py [hunk 2]
```diff
@@ -184,14 +189,28 @@ def cron_automation(self):
...
     def _get_automation_records_to_create(self):
         """
         We will find all the records that fulfill the domain but don't have a record created.
         Also, we need to check by autencity field if defined.
 
         In order to do this, we will add some extra joins on the query of the domain
         """
-        domain = safe_eval(self.domain)
+        eval_context = self._get_eval_context()
+
```
Comment (pedrobaeza) [inline automation_oca/models/automation_configuration.py]: Avoid empty lines inside a method.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The commit message is still incorrect (for both the length of the first line and the tag):

```
[IMP] automation_oca: Allow some expressions in domain

This commit adds the corresponding imports and preparation
code for using some variables and expressions like `today()`,
`datetime`, `user`, etc in the domain (filter) of the automation
configuration.
```

I also miss some explanation of this in the help of the field.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
The help on the field is still missing, but I don't block more.

**Comment by flotho** [comment]
> The help on the field is still missing, but I don't block more.

Thanks @pedrobaeza , @gaelTorrecillas could be proud of this validation.

**Comment by gaelTorrecillas** [comment]
Hello pedro,

thanks you for your reviews.

Regards

**Comment by flotho** [comment]
Ping @DorianMAG @JulienMartinez , thanks for your review.
@gaelTorrecillas  could you produce a usecase example on runboat with a screenshot.

**Comment by pedrobaeza** [comment]
Another option is to include demo data with such domain.

File: automation_oca/models/automation_configuration.py [hunk 3]
```diff
@@ -24,9 +24,25 @@ class AutomationConfiguration(models.Model):
...
-        required=True, default="[]", help="Filter to apply", compute="_compute_domain"
+        required=True,
+        default="[]",
+        help="Filter to apply",
+        compute="_compute_domain",
+        help="""
+        Following special variable can be used in filter :
+         * datetime
+         * dateutil
+         * time """,
```
Comment (pedrobaeza) [inline automation_oca/models/automation_configuration.py]: What about `user` and `ref`?

**Review by pedrobaeza — COMMENTED** [review]
Great changes to complete the module.

File: automation_oca/models/automation_configuration.py (hunk 3 above)
Comment (flotho) [reply automation_oca/models/automation_configuration.py]: you're right @pedrobaeza 
we missed this part, could you have a look @gaelTorrecillas  ?

File: automation_oca/models/automation_configuration.py [hunk 4]
```diff
@@ -19,9 +24,25 @@ class AutomationConfiguration(models.Model):
     tag_ids = fields.Many2many("automation.tag")
     company_id = fields.Many2one("res.company")
     domain = fields.Char(
-        required=True, default="[]", help="Filter to apply", compute="_compute_domain"
+        required=True,
+        default="[]",
+        help="Filter to apply",
```
Comment (flotho) [inline automation_oca/models/automation_configuration.py]: repeated key, impossible to install the module

**Review by flotho — CHANGES_REQUESTED** [review]

**Review by flotho — APPROVED** [review]
tested on runboat, LGTM

**Comment by gaelTorrecillas** [comment]
Hello,

I added demo and user and ref in the fields info.

Regards,

**Pull request merged**